aesara/link/numba/dispatch.py

```diff
@@ -1804,3 +1814,18 @@ def batched_dot(x, y):
 # NOTE: The remaining `aesara.tensor.blas` `Op`s appear unnecessary, because
 # they're only used to optimize basic `Dot` nodes, and those GEMV and GEMM
 # optimizations are apparently already performed by Numba
+
+
+@numba_funcify.register(RandomVariable)
+def numba_funcify_RandomVariable(op, node, **kwargs):
+    name = op.name
+    random_func = getattr(np.random, name)
+
+    @numba.njit
+    def random_variable(rng, size, dtype, *args):
```

Comment:
Got it, makes sense.

Reply:
Writing a manual loop here is find&mdash;if necessary&mdash;because Numba can produce very performant loops.  My main concern is the overhead implied by a large number of independent calls to the underlying sampler code, instead of one call that handles the `size` parameter.  I still need to look at the Numba implementation to get a sense of that overhead, if any.

Regardless, a loop will work for now, and, to handle arbitrary dimensions in such a loop, we can just use `np.ndindex`.